.github/workflows/main.yml

```diff
@@ -83,51 +85,47 @@ jobs:
-        python3 setup.py install
-        pip3 install --upgrade numpy
-        #pip3 install --upgrade rasterio==1.1.8
+        sudo apt-get update
+        DEBIAN_FRONTEND="noninteractive" sudo apt-get install -qy libgdal-dev
+        pdm install --group :all
     - name: setup test data ⚙️
       run: |
-        python3 tests/load_es_data.py tests/data/ne_110m_populated_places_simple.geojson geonameid
-        python3 tests/load_es_data.py tests/cite/ogcapi-features/canada-hydat-daily-mean-02HC003.geojson IDENTIFIER
-        python3 tests/load_mongo_data.py tests/data/ne_110m_populated_places_simple.geojson
+        pdm run python tests/load_es_data.py tests/data/ne_110m_populated_places_simple.geojson geonameid
+        pdm run python tests/load_es_data.py tests/cite/ogcapi-features/canada-hydat-daily-mean-02HC003.geojson IDENTIFIER
+        pdm run python tests/load_mongo_data.py tests/data/ne_110m_populated_places_simple.geojson
         gunzip < tests/data/hotosm_bdi_waterways.sql.gz | psql postgresql://postgres:${{ secrets.DatabasePassword || 'postgres' }}@localhost:5432/test
     - name: run unit tests ⚙️
       env:
         POSTGRESQL_PASSWORD: ${{ secrets.DatabasePassword || 'postgres' }}
       run: |
-        pytest tests/test_api.py
-        pytest tests/test_config.py
-        pytest tests/test_csv__formatter.py
-        pytest tests/test_csv__provider.py
-        pytest tests/test_elasticsearch__provider.py
-        pytest tests/test_esri_provider.py
-        pytest tests/test_filesystem_provider.py
-        pytest tests/test_geojson_provider.py
-        pytest tests/test_mongo_provider.py
-        #pytest tests/test_ogr_csv_provider.py
-        #pytest tests/test_ogr_esrijson_provider.py
-        #pytest tests/test_ogr_gpkg_provider.py
-        #pytest tests/test_ogr_shapefile_provider.py
-        #pytest tests/test_ogr_sqlite_provider.py
-        #pytest tests/test_ogr_wfs_provider.py
-        pytest tests/test_openapi.py
-        pytest tests/test_postgresql_provider.py
-        pytest tests/test_rasterio_provider.py
-        pytest tests/test_sensorthings_provider.py
-        pytest tests/test_socrata_provider.py
-        pytest tests/test_sqlite_geopackage_provider.py
-        pytest tests/test_tinydb_catalogue_provider.py
-        pytest tests/test_util.py
-        pytest tests/test_xarray_netcdf_provider.py
-        pytest tests/test_xarray_zarr_provider.py
+        pdm run pytest tests/test_api.py
+        pdm run pytest tests/test_config.py
+        pdm run pytest tests/test_csv__formatter.py
+        pdm run pytest tests/test_csv__provider.py
+        pdm run pytest tests/test_elasticsearch__provider.py
+        pdm run pytest tests/test_esri_provider.py
+        pdm run pytest tests/test_filesystem_provider.py
+        pdm run pytest tests/test_geojson_provider.py
+        pdm run pytest tests/test_mongo_provider.py
+        #pdm run pytest tests/test_ogr_csv_provider.py
```

Comment:
Not sure why all OGR Provider tests are commented out. Ok, they were already in current workflow.
Except for WFS Provider, these should all run. Maybe give it a try.

Reply:
I think we should invoke pytest just once on all available, making the output more readable, and making it simple to handle the tests. We can skip some of them using the `--ignore` flag.